app/services/apple_auth_service.rb

```diff
@@ -2,22 +2,43 @@
 
 class AppleAuthService < BaseAuthService
   class << self
-    def find_user_by_token(apple_id_token, apple_given_name = nil, apple_family_name = nil)
-      decoded_token = decode_token(apple_id_token)
-      validate_token!(apple_id_token, decoded_token)
-      validate_expected_fields!(decoded_token)
+    def find_user_by_refresh_token(refresh_token)
+      apple_id_client.refresh_token = refresh_token
+      find_user
+    end
+
+    def find_user_by_auth_code(apple_auth_code, apple_given_name = nil, apple_family_name = nil)
+      apple_id_client.authorization_code = apple_auth_code
+      [find_user(apple_given_name, apple_family_name), @response.refresh_token]
+    end
+
+    def find_user(apple_given_name = nil, apple_family_name = nil)
+      @response = apple_id_client.access_token!
+      id_token = @response.id_token
+
+      validate_expected_fields!(id_token.raw_attributes)
+
+      id_token.verify!(
+        client: apple_id_client,
+        access_token: @response.access_token
+      )
 
-      apple_id_token = remote_user_id(decoded_token)
-      user_atts = extract_user_atts(apple_id_token, decoded_token, apple_id_token)
+      user_atts = {apple_user_id: id_token.sub, email: id_token.email}
```

Comment:
Does `id_token.email` really exist? I don't see it in the docs at https://github.com/nov/apple_id/wiki/IdToken

Reply:
`email` is one of the [claims](https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api/authenticating_users_with_sign_in_with_apple#3383773) in the `id_token` actually returned from apple

I would hope/assume that the `apple_id` library just exposes whatever claims get returned from apple.